api.go

```diff
@@ -321,24 +328,43 @@ type RepoListEntry struct {
 	Stats         RepoStats
 }
 
+type MinimalRepoListEntry struct {
+	HasSymbols bool
+	Branches []RepositoryBranch
+}
+
 // RepoList holds a set of Repository metadata.
 type RepoList struct {
+	// Full response to a List request. Returned when ListOptions.Minimal is false.
 	Repos   []*RepoListEntry
+
 	Crashes int
+
+	// Minimal response to a List request. Returned when ListOptions.Minimal is true.
+	Minimal map[uint32]*MinimalRepoListEntry
```

Comment:
why a map? Can't we just use a list? list uses less mem + more efficient to marshal/unmarshal. Also get the benefit of simpler aggregation code, you can just use append.

Reply:
Both in Sourcergaph's side and in shardedSearcher we need to use maps for deduplication and fast lookups. Even for the existing normal `RepoListEntry` we already create an intermediate `map[string]*RepoListEntry` for deduping. On Sourceraph's side, we convert all of that to map again, so any gains in marshalling performance are offset by that.

How much more memory does a map use in Gob encoding?

Reply:
Cool, after more chatting we are gonna forward with this. It shouldn't make that much of a difference, so it's more aesthetic. Additionally the benefit of fast lookups is great on the client side.

Reply:
FYI @keegancsmith, map is slightly smaller in serialized size but a little slower and does allocs. Could likely use a customer Gob encoding function: https://github.com/sourcegraph/zoekt/pull/99/commits/72bcfba482eec95719af3a13e5c1e40628fde080